universal-application-tool-0.0.1/app/assets/javascripts/admin_validation.ts

```diff
@@ -0,0 +1,74 @@
+/** The validation controller provides basic client-side validation of admin form fields. */
+class AdminValidationController {
+  static readonly MULTI_OPTION_QUESTION_FIELD_NAME_CREATE = '#question-settings input[name="newOptions[]"]';
+  static readonly MULTI_OPTION_QUESTION_FIELD_NAME_EDIT = '#question-settings input[name="options[]"]';
+
+  isMultiOptionCreateValid = true;
+  isMultiOptionEditValid = true;
+
+  constructor() {
+    // Attach listener to admin program edit form.
+    const adminProgramEditForm = document.getElementById("question-form");
+    if (adminProgramEditForm) {
+      adminProgramEditForm.addEventListener("submit", (event) => { return this.attemptSubmit(event); });
+    }
+  }
+
+  private attemptSubmit(event: Event): boolean {
+    this.checkFields();
+    if (!this.isValid()) {

```

Comment:
suggestion (non-blocking): recommend removing `isValid` and having `checkFields` (a.k.a. `validateAll`) return a boolean. `isValid` scares me a little bit because it relies on another method getting called first for it its return value to be correct.

Reply:
That's cleaner, good suggestion. Fixed.